src/main/drivers/max7456.c

```diff
@@ -669,13 +676,34 @@ void max7456DrawScreen(void)
         int buff_len = 0;
         for (int k = 0; k < MAX_CHARS2UPDATE; k++) {
             if (screenBuffer[pos] != shadowBuffer[pos]) {
-                spiBuff[buff_len++] = MAX7456ADD_DMAH;
-                spiBuff[buff_len++] = pos >> 8;
-                spiBuff[buff_len++] = MAX7456ADD_DMAL;
-                spiBuff[buff_len++] = pos & 0xff;
-                spiBuff[buff_len++] = MAX7456ADD_DMDI;
-                spiBuff[buff_len++] = screenBuffer[pos];
-                shadowBuffer[pos] = screenBuffer[pos];
+                if (screenBuffer[pos] <= 255) {
+                    spiBuff[buff_len++] = MAX7456ADD_DMAH;
+                    spiBuff[buff_len++] = pos >> 8;
+                    spiBuff[buff_len++] = MAX7456ADD_DMAL;
+                    spiBuff[buff_len++] = pos & 0xff;
+                    spiBuff[buff_len++] = MAX7456ADD_DMDI;
+                    spiBuff[buff_len++] = screenBuffer[pos];
+                    shadowBuffer[pos] = screenBuffer[pos];
+                } else {
+                    spiBuff[buff_len++] = MAX7456ADD_DMM;
+                    spiBuff[buff_len++] = 64;
+                    spiBuff[buff_len++] = MAX7456ADD_DMAH;
+                    spiBuff[buff_len++] = (pos >> 8) | 0x2;
+                    spiBuff[buff_len++] = MAX7456ADD_DMAL;
+                    spiBuff[buff_len++] = pos & 0xff;
+                    spiBuff[buff_len++] = MAX7456ADD_DMDI;
+                    spiBuff[buff_len++] = 0x10;
+                    spiBuff[buff_len++] = MAX7456ADD_DMAH;
+                    spiBuff[buff_len++] = pos >> 8;
+                    spiBuff[buff_len++] = MAX7456ADD_DMDI;
+                    spiBuff[buff_len++] = screenBuffer[pos] & 0xff;
+                    // Put DMM back
+                    spiBuff[buff_len++] = MAX7456ADD_DMM;
+                    spiBuff[buff_len++] = displayMemoryModeReg;
+                    shadowBuffer[pos] = screenBuffer[pos];
+                    //Increment twice (14 bytes is equiv to 3 proper chars)
+                    k++;k++;
```

Comment:
This will waste 4 bytes per extended character. It may be better to check `buff_len` before writing next character